letsencrypt/cli.py

```diff
@@ -1282,19 +1294,36 @@ def __call__(self, parser, config, webroot, option_string=None):
                                "them must precede all domain flags")
         config.webroot_path.append(webroot)
 
+_undot = lambda domain: domain[:-1] if domain.endswith('.') else domain
+
+def _process_domain(config, domain_arg, webroot_path=None):
+    """
+    Process a new -d flag, helping the webroot plugin construct a map of
+    {domain : webrootpath} if -w / --webroot-path is in use
+    """
+    webroot_path = webroot_path if webroot_path else config.webroot_path
+
+    for domain in (d.strip() for d in domain_arg.split(",")):
+        if domain not in config.domains:
+            domain = _undot(domain)
+            config.domains.append(domain)
+            # Each domain has a webroot_path of the most recent -w flag
+            # unless it was explicitly included in webroot_map
+            if webroot_path:
+                config.webroot_map.setdefault(domain, webroot_path[-1])
+
+
+class WebrootMapProcessor(argparse.Action): # pylint: disable=missing-docstring
+    def __call__(self, parser, config, webroot_map_arg, option_string=None):
+        webroot_map = json.loads(webroot_map_arg)
+        for domains, webroot_path in webroot_map.iteritems():
+            _process_domain(config, str(domains), [str(webroot_path)])
```

Comment:
Right now something like `-d żąćł.com` results in a nice error "Internationalized domain names are not presently supported". `--webroot-map "{'żąćł.com': '/tmp'}"` on the other hand raises `UnicodeEncodeError`. Would be nice to have consistency between these two options.
Maybe moving `str(domains)` to `_process_domain` and wrapping it in a try clause?

`str(webroot_path)` raises an uncaught exception here as well, but utf-8 characters in webroot_path seem to be unsupported now anyway?